components/layout/fragment.rs

```diff
@@ -2482,26 +2482,49 @@ impl Fragment {
     pub fn meld_with_next_inline_fragment(&mut self, next_fragment: &Fragment) {
         if let Some(ref mut inline_context_of_this_fragment) = self.inline_context {
             if let Some(ref inline_context_of_next_fragment) = next_fragment.inline_context {
-                for (i, inline_context_node_from_next_fragment) in
-                        inline_context_of_next_fragment.nodes.iter().enumerate() {
-                    if i >= inline_context_of_this_fragment.nodes.len() {
-                        continue
-                    }
+                for (inline_context_node_from_this_fragment,
+                     inline_context_node_from_next_fragment)
+                    in inline_context_of_this_fragment.nodes.iter_mut().rev()
+                        .zip(inline_context_of_next_fragment.nodes.iter().rev())
+                {
                     if !inline_context_node_from_next_fragment.flags.contains(
                             LAST_FRAGMENT_OF_ELEMENT) {
                         continue
                     }
                     if inline_context_node_from_next_fragment.address !=
-                            inline_context_of_this_fragment.nodes[i].address {
+                            inline_context_node_from_this_fragment.address {
                         continue
                     }
-                    inline_context_of_this_fragment.nodes[i].flags.insert(
-                        LAST_FRAGMENT_OF_ELEMENT);
+                    inline_context_node_from_this_fragment.flags.insert(LAST_FRAGMENT_OF_ELEMENT);
                 }
             }
         }
     }
 
+    pub fn meld_with_prev_inline_fragment(&mut self, prev_fragment: &Fragment) {
+        if let Some(ref mut inline_context_of_this_fragment) = self.inline_context {
+            if let Some(ref inline_context_of_prev_fragment) = prev_fragment.inline_context {
```

Comment:
Tried out this suggestion, but ended up making the code less readable because of explicit reborrows needed to avoid moving from borrowed content into the tuple.